### DISCUSSION TITLE
Can I use @chakra-ui/layout with other UI library like antd?

### DISCUSSION THREAD

**Discussion by nshen** [chakra-ui/chakra-ui#4947] [Q&A] [answered]
chakra-ui is a great styled-components library, but it lacks some advanced components.
Is that possible to use chakra-ui just for layout and with other UI components libraries use together?

**Comment by simonsankar** [comment]
I too find myself in this dilemma. ChakraUI is great for cosmetics + layout fine-tuning but lacks robust components like antd.
Is there a way to just setup chakraui for solely layouts and use other react based components

**Comment by ChristopherWirtOfficial** [comment, marked as answer]
In terms of "can I? or "is there a way to" the answer is a resounding YES. There's nothing stopping you from using a normal installation of Chakra and using it just for the styled components. You can even do this without setting up a custom theme, or set a theme that inherits from some other theme you're using.

For the question of "Should I?", that's a different story. There's nothing inherently wrong in terms of compatibility, but there are some obvious maintenance concerns with mixing two UI Frameworks. If you're diligent and have good guidelines for when to use ChakraUI vs your other UI framework, it can be a very good solution. However, if you blur the lines on which one you use for what things, or have lots of legacy code that is built without Chakra, you may be in for some rough maintenance problems.
